src/components/__tests__/App.test.tsx

```diff
@@ -0,0 +1,110 @@
+/// <reference types="vitest/globals" />
+import { render, screen, waitFor } from '@testing-library/react';
+import App from '../../App';
+import userEvent from '@testing-library/user-event';
+import ErrorBoundary from '../ErrorBoundary';
+
+beforeEach(() => {
+  vi.stubGlobal('fetch', vi.fn());
+  localStorage.clear();
+});
+
+afterEach(() => {
+  vi.restoreAllMocks();
+});
+
+describe('App Component', () => {
+  it('loads and displays Pokemon list when no search term in localStorage', async () => {
+    const mockedListResponse = {
+      results: [
+        { name: 'bulbasaur', url: 'https://pokeapi.co/api/v2/pokemon/1/' },
+      ],
+    };
+    const mockPokemonDetail = {
+      name: 'bulbasaur',
+      height: 7,
+      weight: 69,
+    };
+    const globalFetch = fetch as unknown as ReturnType<typeof vi.fn>;
+    globalFetch
+      .mockResolvedValueOnce({ ok: true, json: async () => mockedListResponse })
+      .mockResolvedValueOnce({ ok: true, json: async () => mockPokemonDetail });
+
+    render(<App></App>);
```

Comment:
JSX provides ability to use shorthands as 
```suggestion
    render(<App />);
```

Reply:
Done! Used shorthand in every test case.